src/afrolabs/components/kafka/utilities.clj

```diff
@@ -29,18 +29,25 @@
            [org.apache.kafka.clients.admin ListTopicsOptions]))
 
 (defn load-messages-from-confluent-topic
-  "Loads a collection of messages from confluent kafka topics. Will stop consuming when the consumer has reached the very latest offsets.
+  "Loads a collection of messages from confluent kafka topics. Will stop consuming when the consumer has reached the
+   very latest offsets.
 
   Intended for interactive use. Returns items in an unspecified order.
 
   - Specify :topics (a collection of strings) for which topics to subscribe to.
-  - If you want to limit the number of messages you load, use :nr-msgs to specify a lower bound of loaded messages. You will get at least this many, unless there are too few messages in the topic, in which case you will get everything in the topic.
+  - If you want to limit the number of messages you load, use :nr-msgs to specify a lower bound of loaded messages.
+    You will get at least this many, unless there are too few messages in the topic, in which case you will get
+    everything in the topic.
   - Use the :msg-filter to keep a subset of messages of the ones in the topic(s)
   - Optionally specify consumer-group-id, otherwise a fresh and unique consumer-group-id will be used.
   - Optionally specify :offset-reset (one of #{\"earliest\" \"latest\"}. Default \"earliest\".)
   - ConfluentCloud specific: when :api-key and :api-secret are both nil, the ConfluentCloud strategy will not be used.
+  - You may optionally set `collect-messages?` to false, in which case no records will be returned. This is useful
+    in streaming mode.
+  - You may specify `stream-ch`, which will accept a collection of kafka messages. This is useful in conjunction with
+    setting `collect-messages?` to `false`. NOTE: `stream-ch` will be closed when consumption is done.
```

Comment:
These were useful to load a kafka topic into a `tech.ml.dataset`, which is extremely {space,processing}-efficient. Loading ~2.5M maps into a vector gave me OutOfMemoryExceptions.